File: android/src/main/kotlin/io/openim/flutter_openim_sdk_ffi/listener/OnFriendshipListener.kt
package io.openim.flutter_openim_sdk_ffi.listener

import io.openim.flutter_openim_sdk_ffi.models.BlacklistInfo
import io.openim.flutter_openim_sdk_ffi.models.FriendApplicationInfo
import io.openim.flutter_openim_sdk_ffi.models.FriendInfo

/**
 * 好友关系监听
 */
interface OnFriendshipListener {
    fun onBlacklistAdded(u: BlacklistInfo)
    fun onBlacklistDeleted(u: BlacklistInfo)
    fun onFriendApplicationAccepted(u: FriendApplicationInfo)
    fun onFriendApplicationAdded(u: FriendApplicationInfo)
    fun onFriendApplicationDeleted(u: FriendApplicationInfo)
    fun onFriendApplicationRejected(u: FriendApplicationInfo)
    fun onFriendInfoChanged(u: FriendInfo)
    fun onFriendAdded(u: FriendInfo)
    fun onFriendDeleted(u: FriendInfo)
}

File: ios/Classes/listener/OnFriendshipListener.swift


/**
 * 好友关系监听
 */
protocol OnFriendshipListener {
    func onBlacklistAdded(_ info: BlacklistInfo)
    func onBlacklistDeleted(_ info: BlacklistInfo)
    func onFriendApplicationAccepted(_ info: FriendApplicationInfo)
    func onFriendApplicationAdded(_ info: FriendApplicationInfo)
    func onFriendApplicationDeleted(_ info: FriendApplicationInfo)
    func onFriendApplicationRejected(_ info: FriendApplicationInfo)
    func onFriendInfoChanged(_ info: FriendInfo)
    func onFriendAdded(_ info: FriendInfo)
    func onFriendDeleted(_ info: FriendInfo)
}